src/data/mv_polynomial/derivation.lean

```diff
@@ -0,0 +1,122 @@
+/-
+Copyright (c) 2021 Yury Kudryashov. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yury Kudryashov
+-/
+import data.mv_polynomial.basic
+import ring_theory.derivation
+
+/-!
+# Derivations of multivariate polynomials
+
+In this file we prove that a variation of `mv_polynomial σ R` is defined by its values on all
+monomials `mv_polynomial.X i`. We also provide a constructor `mv_polynomial.mk_derivation` that
+builds a derivation from its values on `X i`s and a linear equivalence
+`mv_polynomial.equiv_derivation` between `σ → A` and `derivation (mv_polynomial σ R) A`. -/
+
+namespace mv_polynomial
+
+open_locale big_operators
+
+noncomputable theory
+
+variables {σ R A : Type*} [comm_ring R] [add_comm_group A]
+  [module R A] [module (mv_polynomial σ R) A]
+
+section
+
+variable (R)
+
+def mk_derivationₗ (f : σ → A) :
+  mv_polynomial σ R →ₗ[R] A :=
```

Comment:
```suggestion
def mk_derivationₗ (f : σ → A) : mv_polynomial σ R →ₗ[R] A :=
```
suggesting +committing this only because the cache for this PR seems corrupt, and I need to change something to force it to rebuild